Source/Login/Backend/configuration/viper/providers_configuration.go

```diff
@@ -0,0 +1,71 @@
+package viper
+
+import (
+	"net/url"
+
+	"dolittle.io/login/providers"
+	"github.com/spf13/viper"
+)
+
+const (
+	providersKey         = "providers"
+	providersNameKey     = "name"
+	providersImageURLKey = "image_url"
+)
+
+var (
+	defaultProviderName     = "UNKNOWN PROVIDER"
+	defaultProviderImageURL = ""
+)
+
+func newNilProvider() *providers.ProviderConfiguration {
+	return &providers.ProviderConfiguration{
+		Name:     defaultProviderName,
+		ImageURL: nil,
+	}
+}
+
+type providersConfiguration struct{}
+
+func (c *providersConfiguration) Providers() providers.Providers {
+	if !viper.IsSet(providersKey) {
+		return nil
+	}
+	return getProviderConfigurationMap(viper.GetStringMap(providersKey))
+}
+
+func getProviderConfigurationMap(in map[string]interface{}) providers.Providers {
+	providerConfigMap := make(map[providers.ProviderID]*providers.ProviderConfiguration, len(in))
```

Comment:
I don't think we need to be fancy to save memory allocations with the size:
```suggestion
	providerConfigMap := make(map[providers.ProviderID]*providers.ProviderConfiguration)
```

Reply:
👍

Reply:
Fixed by https://github.com/dolittle-platform/Authentication/pull/10/commits/68d3ca5afdfe219c74055ab494857bcdec2339be